job/cache.go

```diff
@@ -115,6 +115,18 @@ func (c *MemoryJobCache) Delete(id string) {
 	c.jobs.Lock.Lock()
 	defer c.jobs.Lock.Unlock()
 
+	j := c.jobs.Jobs[id]
+	if j == nil {
+		// TODO
+		//return nil, JobDoesntExistErr
+		return
+	}
+
+	// TODO return error
+	c.jobs.Lock.Unlock()
```

Comment:
@ajvb It seems like Job deletion and a Job being deleted from its parent Jobs should be atomic, but it's not; there are separate locks for each, and they are not locked and unlocked together.

You could have a 3rd lock that ensures atomicity for you, but that's just getting ridiculous.

Simplify the design!

Thoughts, @samdolan?